metaflow/task.py

```diff
@@ -493,6 +512,36 @@ def run_step(
                     )
                 )
 
+            # Add runtime dag info - for a nested foreach this may look like:
```

Comment:
in terms of metadata that needs to be stored - we could simply stringify the stack and store it - `step1:0,step2:1...` - and that should be it, no? a prefix query should be fast enough